reproducible_build.sh

```diff
@@ -56,74 +80,258 @@ if [ "$CLONE" = false ]; then
     SOURCE_DIR=$PWD
 fi
 
-if [ -z ${TRAVIS+x} ]; then
-  # if not on travis, its nice to see progress
-  QUIET=""
+if [ -z ${TRAVIS_OS_NAME+x} ]; then
+    if [ `uname -s` = "Darwin" ]; then
+	OS_NAME="osx"
+    else
+	OS_NAME="linux"
+    fi
 else
-    QUIET="-qq"
+    OS_NAME=${TRAVIS_OS_NAME}
 fi
 
-# get the required OS packages
-$SUDO apt-get ${QUIET} update
-$SUDO apt-get ${QUIET} install -y --no-install-recommends \
-    build-essential python-dev libbz2-dev libtool \
-    autotools-dev autoconf git pkg-config wget \
-    ca-certificates automake bsdmainutils
-
-mkdir dependencies
-cd dependencies
-
-wget http://download.oracle.com/berkeley-db/db-4.8.30.NC.tar.gz
-tar xf db-4.8.30.NC.tar.gz
-export BDB_PREFIX="`pwd`/bdb"
-cd db-4.8.30.NC/build_unix
-../dist/configure --prefix=$BDB_PREFIX --enable-cxx --disable-shared --with-pic
-make
-make install
-cd ../../
-
-wget https://www.openssl.org/source/openssl-1.0.1p.tar.gz
-tar xf openssl-1.0.1p.tar.gz
-export OPENSSL_PREFIX="`pwd`/openssl_build"
-mkdir $OPENSSL_PREFIX
-cd openssl-1.0.1p
-./Configure --prefix=$OPENSSL_PREFIX --openssldir=$OPENSSL_PREFIX/ssl linux-x86_64 -fPIC -static no-shared no-dso
-make
-make install
-cd ..
-
-
-wget http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download -O boost_1_59_0.tar.bz2
-tar xf boost_1_59_0.tar.bz2
-export BOOST_ROOT="`pwd`/boost_1_59_0"
-cd boost_1_59_0
-./bootstrap.sh
-./b2 link=static cxxflags=-fPIC stage
-cd ../../
-
-
-mkdir libevent_build
-git clone https://github.com/libevent/libevent.git
-export LIBEVENT_PREFIX="`pwd`/libevent_build"
-cd libevent
-./autogen.sh
-./configure --prefix=$LIBEVENT_PREFIX --enable-static --disable-shared --with-pic LDFLAGS="-L${OPENSSL_PREFIX}/lib/" CPPFLAGS="-I${OPENSSL_PREFIX}/include"
-make
-make install
-cd ..
+NEXT_TIME=60
+function exit_at_45() {
+    if [ -f ${HOME}/start_time ]; then
```

Comment:
Regarding the trap, I like having it set for that dependency so that it doesn't delete the other, already completed dependencies.  I don't need to chain them because once a dependency is built I can, and do, remove the trap associated with that one.

I'll fix up the ${HOME}/start_time bit.